File: client/game_card/Assets/Scripts/App/AppStates/AppStateWaitTask.cs

using System;
using Utopia;

namespace Utopia
{
    public class AppStateWaitTask : AppStateBase
    {
        Action m_task = null;

        public AppStateWaitTask(AppStateMgr stateMgr) : base(stateMgr, EAppState.WaitTask)
        {

        }

        public override void Enter(object param)
        {

        }

        public override void Exit()
        {

        }

        public override void Update()
        {
            if (null != m_task)
            {
                Action task = m_task;
                m_task = null;
                task();
            }
        }

        public void SetTask(Action task)
        {
            m_task = task;
        }
    }
}

File: client/game_card/LuaScript/app/state_mgr/app_state/app_state_wait_task.lua

AppStateWaitTask = AppStateWaitTask or class("AppStateWaitTask", AppStateBase)

function AppStateWaitTask:ctor(state_mgr, main_logic)
    AppStateWaitTask.super.ctor(self, state_mgr, App_State_Name.wait_task, main_logic)
end

